### PR TITLE
F/fix osx dup symbols

### LINKED ISSUE #807: Duplicate symbols (_TLSSslOptions and _last_parser) break build of syslog-ng 3.7.2 on OS X
Setup: OS X 10.11, Apple LLVM version 7.0.0 (clang-700.1.76)

After bypassing the issue mentioned in issue #806 by just making check_nanosleep() in timeutils.c return FALSE, I ran into a 'duplicate symbols' error when linking libsyslog-ng:
`````
Douglass-Mac-Pro:syslog-ng-3.7.2 dcarmich$ make
/Applications/Xcode.app/Contents/Developer/usr/bin/make --no-print-directory all-recursive
  CCLD     lib/libsyslog-ng.la
clang: warning: argument unused during compilation: '-pthread'
clang: warning: argument unused during compilation: '-pthread'
duplicate symbol _TLSSslOptions in:
    lib/.libs/lib_libsyslog_ng_la-tlscontext.o
    lib/transport/.libs/lib_libsyslog_ng_la-transport-tls.o
duplicate symbol _last_parser in:
    lib/.libs/lib_libsyslog_ng_la-cfg-grammar.o
    lib/parser/.libs/lib_libsyslog_ng_la-parser-expr-grammar.o
ld: 2 duplicate symbols for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/libsyslog-ng.la] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```````

### LINKED ISSUE #808: OS X ld does not support --whole-archive option, thus breaking the build of 3.7.2 on OS X
The first issue I dealt with trying to build 3.7.2 on OS X is that ld on OS X does not support --whole-archive.

There is a patch to fix it as part of [MacPorts ticket #43231](https://trac.macports.org/attachment/ticket/43231/patch-configure.ac.diff). Should this be applied to the main repository?

### PR DESCRIPTION
**Pull request by ihrwein**
Fixes #807 (I hope)

### CODE REVIEW

File: lib/parser/parser-expr-grammar.ym [hunk 1]
```diff
@@ -36,8 +36,6 @@
 #include "plugin.h"
 #include "cfg-grammar.h"
 
-LogParser *last_parser;
-
 }
```
Comment (bazsi) [inline lib/parser/parser-expr-grammar.ym]: I think this should be the canonical definition. Where else is this defined?

Comment (ihrwein) [inline lib/parser/parser-expr-grammar.ym]: I suppose the definition comes from `cfg-grammar.y`.

**Comment by bkil-syslogng** [comment]
Duplicates #938. Please consider closing this one if you like that one.

**Comment by bazsi** [comment]
Closing as requested.

**Pull request closed without merging**